pkg/querier/querier_test.go

```diff
@@ -686,6 +686,328 @@ func TestQuerier_buildQueryIntervals(t *testing.T) {
+		expectedCallsIngesters           int
+	}{
+		{
+			desc:                   "query both storage and ingesters",
+			start:                  time.Now().Add(-time.Hour * 2),
+			end:                    time.Now(),
+			expectedCallsStore:     1,
+			expectedCallsIngesters: 1,
+		},
+		{
+			desc:                   "query ingesters only (IngesterQueryStoreMaxLookback not set)",
+			start:                  time.Now().Add(-time.Minute * 15),
+			end:                    time.Now(),
+			expectedCallsStore:     1,
+			expectedCallsIngesters: 1,
+		},
+		{
+			desc:                   "query storage only",
+			start:                  time.Now().Add(-time.Hour * 2),
+			end:                    time.Now().Add(-time.Hour * 1),
+			expectedCallsStore:     1,
+			expectedCallsIngesters: 0,
+		},
+		{
+			desc:                             "query ingesters only (IngesterQueryStoreMaxLookback set)",
+			start:                            time.Now().Add(-time.Minute * 15),
+			end:                              time.Now(),
+			setIngesterQueryStoreMaxLookback: true,
+			expectedCallsStore:               0,
+			expectedCallsIngesters:           1,
+		},
+	}
+
+	requests := []struct {
+		name string
+		do   func(querier *SingleTenantQuerier, start, end time.Time) error
+	}{
+		{
+			name: "SelectLogs",
+			do: func(querier *SingleTenantQuerier, start, end time.Time) error {
+				_, err := querier.SelectLogs(ctx, logql.SelectLogParams{
+					QueryRequest: &logproto.QueryRequest{
+						Selector:  "{type=\"test\", fail=\"yes\"} |= \"foo\"",
+						Limit:     10,
+						Start:     start,
+						End:       end,
+						Direction: logproto.FORWARD,
+					},
+				})
+
+				return err
+			},
+		},
+		{
+			name: "SelectSamples",
+			do: func(querier *SingleTenantQuerier, start, end time.Time) error {
+				_, err := querier.SelectSamples(ctx, logql.SelectSampleParams{
+					SampleQueryRequest: &logproto.SampleQueryRequest{
+						Selector: "count_over_time({foo=\"bar\"}[5m])",
+						Start:    start,
+						End:      end,
+					},
+				})
+				return err
+			},
+		},
+		{
+			name: "LabelValuesForMetricName",
+			do: func(querier *SingleTenantQuerier, start, end time.Time) error {
+				_, err := querier.Label(ctx, &logproto.LabelRequest{
+					Name:   "type",
+					Values: true,
+					Start:  &start,
+					End:    &end,
+				})
+				return err
+			},
+		},
+		{
+			name: "LabelNamesForMetricName",
+			do: func(querier *SingleTenantQuerier, start, end time.Time) error {
+				_, err := querier.Label(ctx, &logproto.LabelRequest{
+					Values: false,
+					Start:  &start,
+					End:    &end,
+				})
+				return err
+			},
+		},
+		{
+			name: "Series",
+			do: func(querier *SingleTenantQuerier, start, end time.Time) error {
+				_, err := querier.Series(ctx, &logproto.SeriesRequest{
+					Start: start,
+					End:   end,
+				})
+				return err
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.desc, func(t *testing.T) {
+
+			conf := mockQuerierConfig()
```

Comment:
```
if tc.setIngesterQueryStoreMaxLookback {
				conf.IngesterQueryStoreMaxLookback = conf.QueryIngestersWithin
			}
```
Missed the `tc.setIngesterQueryStoreMaxLookback`. So that block needs to go inside the outer loop.

```
limits, err := validation.NewOverrides(defaultLimitsTestConfig(), nil)
					require.NoError(t, err)
```
Can go in the outer loop as well